packages/core/src/lib/toast/__tests__/toast.spec.ts

```diff
@@ -35,6 +36,74 @@ describe('toast', () => {
     expect(icon).toHaveClass('text-success-dark');
   });
 
+  it('should display a toast with info variant with correct classes and label', async () => {
```

Comment:
For future reference, this is a good use case for [`it.each`](https://vitest.dev/api/#test-each). I also think the button and message checking is redundant with other tests. Try to minimize redundant coverage and focus in this test on checking the icon

```ts
it.each([
  {
    variant: ToastVariant.SUCCESS,
    expectedName: /success/iu,
    expectedColor: 'text-success-dark',
  },
  // ...
])('displays correct icon for %variant', ({
  variant,
  expectedName,
  expectedColor,
}) => {
  await act(() => context.toast({ message: 'Hello world' }));

  const status = screen.getByRole('status');
  const toast = within(status).getByRole('listitem');
  const icon = within(toast).getByRole('img', { name: expectedName });

  expect(icon).toHaveClass(expectedColor);
})
```